types/luxon/src/datetime.d.ts

```diff
@@ -740,6 +740,9 @@ export class DateTime {
      */
     static isDateTime(o: unknown): o is DateTime;
 
+    // Constructor is declared `@access private`
+    private constructor();
```

Comment:
Sounds good to me.  My initial goal with this PR was to address https://github.com/moment/luxon/issues/900 and marking it protected still does that.

It might be interesting to ask @icambron whether subclassing Luxon classes is "officially supported", or at least encouraged vs discouraged.  I noticed that JSDoc has a `@protected` tag but it pertains to "module-level access" as opposed to ES6 class inheritance; I don't know if JSDoc has the vocabulary to actually express that.

Reply:
Sub-classing is definitely not officially supported. However, I think this strikes an acceptable balance between discouraged but allowed. @peterblazejewicz any thoughts?

Reply:
I definitely don't support subclassing, but I'm generally of the mind that you can and should do whatever you want, as long as you're aware that I might break it in some future release of Luxon. I'm certainly not the boss of you. IMO, though, with Typescript, the right way to handle that is probably to mark it as private and then you `// @ts-ignore` that

Reply:
I'll concede to that. `protected` does seem to encourage extension maybe too much.
Let's at least add the `config` arg to the constructors, so only one error is required to ignore.
```ts
export class CalendarDate
  // @ts-expect-error constructor is private
  extends DateTime {

  private constructor () {
    super({ ... }); // ok
  }
}
```

Reply:
I'd add `@private` (when applicable) and yes, add proper params, where expected. I assume this is for folks like me using this staff in other project, just consuming documented api (and thanks to TS avoiding common coding problems)

Reply:
@peterblazejewicz You'd add a `@private` JSDoc tag even though the TS `private` keyword declares the same thing?

Reply:
The JSDoc tag is for the docs, the TS keyword is to keep library consumers from calling the method.  The one doesn't directly impact the other.  (Of course the project *could* migrate to `tsd-jsdoc` and pack in their own typings!  Which would be great!  But until then, they're not connected...)

Reply:
Oh, no, I usually like (and ask) to add JSDoc where it really helps, say for `@async`, which is not support as keyword in declaration types, `private` should be self documenting